### PR TITLE
add a new command on order items-cancelation

### PR DESCRIPTION
**Pull request by caberos**
[#issue1927](https://github.com/softlayer/softlayer-python/issues/1927)

image

### CODE REVIEW

**Comment by edsonarios** [comment]
Screenshots of the test result
image

**Review by edsonarios — APPROVED** [review]
Looks good to me

File: SoftLayer/CLI/order/cancelation.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+from SoftLayer.CLI.command import SLCommand as SLCommand
+from SoftLayer.CLI import environment
+from SoftLayer.CLI import formatting
+from SoftLayer.managers import ordering
+
+
+@click.command(cls=SLCommand)
+@environment.pass_env
+def cli(env):
+    """List all active quotes on an account"""
```
Comment (edsonarios) [inline SoftLayer/CLI/order/cancelation.py:14]: Maybe is needed to update this description

File: SoftLayer/managers/ordering.py [hunk 2]
```diff
@@ -745,3 +746,14 @@ def delete_quote(self, quote_id):
         """
 
         return self.client['SoftLayer_Billing_Order_Quote'].deleteQuote(id=quote_id)
+
+    def get_all_cancelation(self):
+        """returns the all cancelations, completed orders"""
+
+        mask = 'mask[id,itemCount,modifyDate,createDate,ticketId,' \
```
Comment (allmightyspiff) [inline SoftLayer/managers/ordering.py:753]: Have the mask be only the columns we need to display, which will make the API command much faster.

File: SoftLayer/managers/ordering.py [hunk 3]
```diff
@@ -745,3 +746,14 @@ def delete_quote(self, quote_id):
...
+               'ticket[assignedUserId,id,attachedHardware[id,hostname,domain],' \
+               'attachedVirtualGuests[id,hostname,domain],' \
+               'attachedDedicatedHosts[id,name],serviceProviderResourceId],' \
+               'status[name,id],user[id,firstName,lastName],' \
+               'items[billingItem[cancellationDate,categoryCode,pendingCancellationFlag]]]'
+        return self.client.call('SoftLayer_Billing_Item_Cancellation_Request', 'getAllCancellationRequests', mask=mask)
```
Comment (allmightyspiff) [inline SoftLayer/managers/ordering.py:759]: Add a limit of 50 here, for this method the default is only 25 which I don't think is quite enough. Showing the last 50 cancellations should be good though.

**Review by allmightyspiff — CHANGES_REQUESTED** [review]

**Comment by ramkishor-ch** [comment]
Hi,

I have "tested the changes".
Please find the screenshot for reference.

Screen Shot: 1
Screenshot 2023-05-09 at 4 20 23 PM

Thanks,
Ramkishor Chaladi.

**Review by ramkishor-ch — APPROVED** [review]
Hi,
I have "tested the changes".
Approving the Pull Request.
Thanks,
Ramkishor Chaladi.

**Review by allmightyspiff — APPROVED** [review]

**Pull request merged**